n3fit/src/evolven3fit/evolve.py

```diff
@@ -106,18 +97,51 @@ def evolve_fit(
         else:
             raise ValueError(f"dump_eko not provided and {eko_path=} not found")
 
-    with eko.EKO.edit(eko_path) as eko_op:
-        x_grid_obj = eko.interpolation.XGrid(x_grid)
-        eko.io.manipulate.xgrid_reshape(eko_op, targetgrid=x_grid_obj, inputgrid=x_grid_obj)
-
+    # Open the EKO in read-only mode, if it needs to be manipulated keep it in memory
     with eko.EKO.read(eko_path) as eko_op:
-        # Read the cards directly from the eko to make sure they are consistent
+        # Read the cards directly fro"m the eko to make sure they are consistent
         theory = eko_op.theory_card
         op = eko_op.operator_card
         # And dump them to the log
         _logger.debug(f"Theory card: {json.dumps(theory.raw)}")
         _logger.debug(f"Operator card: {json.dumps(op.raw)}")
 
+        eko_original_xgrid = eko_op.xgrid
+        if XGrid(x_grid) != eko_original_xgrid:
+            # If the xgrid of the eko is not directly usable, construct a copy in memory
+            # by replacing the internal inventory of operators in a readonly copy
+            new_xgrid = XGrid(x_grid)
+            new_metadata = dataclasses.replace(eko_op.metadata, xgrid=new_xgrid)
+
+            new_operators = {}
+            for target_key in eko_op.operators:
```

Comment:
what is `target_key` actually? the "canonical" way is to directly iterate `eko_op` (I would say), which should yield `ep -> elem`

Reply:
This is needed to construct a rotated copy of the original grid in memory, since I need to replace the `cache` variable of the operator.

Reply:
not sure I understand, but if it works - fine. As said maybe something for later improvement ...

Reply:
I see.... but this way you loose the memory management of eko, as you are holding again all EKOs in memory, which for n3fit might still be a significant amount (do you know how many?). Again, if it works let's postpone improvements

Reply:
Take into account that this is almost never used.